core/src/main/java/org/openconnectors/connect/SourceConnector.java

```diff
@@ -0,0 +1,15 @@
+package org.openconnectors.connect;
+
+import org.openconnectors.util.SourceConnectorContext;
+
+import java.util.Collection;
+
+public abstract class SourceConnector<T extends SourceConnectorContext, U> extends Connector<T> {
+
+    public abstract Collection<U> poll() throws Exception;
+
+    public abstract void start(SourceCollector<U> collector) throws Exception;
```

Comment:
This looks good. Only thing is the name SourceCollector is a bit jarring.

Reply:
Well it's a general collector to has have some specificity.

Reply:
Actually I see it now. BTW dont we need some kind of context?

Reply:
SourceCollector may benefit from a context variable but I think it will complicate things too much